src/renderer/captureTool/js/capture.js

```diff
@@ -0,0 +1,1014 @@
+                funcName: "gpii.captureTool.updateInstalledSolutionsSchemas",
+                args: ["{that}"]
+            },
+            onSolutionHeaderToKeepCheck: {
+                funcName: "gpii.captureTool.onSolutionHeaderToKeepCheck",
+                args: ["{that}", "{arguments}.0"]
+            },
+            onHideShowSolution: {
+                funcName: "gpii.captureTool.onHideShowSolution",
+                args: ["{that}", "{arguments}.0"]
+            },
+            onHideShowNumSettingLink: {
+                funcName: "gpii.captureTool.onHideShowNumSettingLink",
+                args: ["{that}", "{arguments}.0"]
+            },
+            onSelectPrefssetDropdown: {
+                funcName: "gpii.captureTool.onSelectPrefssetDropdown",
+                args: ["{that}", "{arguments}.0"]
+            }
+        }
+    });
+
+    gpii.captureTool.fullChange = function (applier, path, value) {
+        var transaction = applier.initiate();
+        transaction.fireChangeRequest({ path: path, value: {}, type: "DELETE"});
+        transaction.fireChangeRequest({ path: path, value: value});
+        transaction.commit();
+    };
+
+    /*
+     * Annotates the merged capture with data for rendering the list including
+     * solution names and the number of settings captured for each solutions.
+     *
+     * Edits the mergedCapture in place, and also re-returns it.
+     */
+    gpii.captureTool.annotateSettingsCapture = function (that, mergedCapture) {
+        var togo = {};
+        fluid.each(mergedCapture, function (capturedSolution, solutionID) {
+            if (!that.model.installedSolutions[solutionID]) {
+                console.log("The solutionID is missing for this installedSolution: ", solutionID);
+                return;
+            }
+
+            togo[solutionID] = {
+                settings: capturedSolution,
+                name: that.model.installedSolutions[solutionID].name,
+                numberOfSettings: Object.keys(capturedSolution).length
+            };
+        });
+        return togo;
+    };
+
+    /*
+     * Takes an object of values keyed by solution ids, converts them in to a list
+     * adding the ID as a value with key `id`, and then sorts them based on our
+     * preferred list of solutions orders for display.
+     *
+     * @param {Object} data - An object with keyed entries, were each key is the id
+     * of a solutions registry entry.
+     */
+    gpii.captureTool.createArrayFromSolutionsHash = function (data) {
+        var orderedInstalledSolutions = [];
+        fluid.each(data, function (val, key) {
+            var next = fluid.copy(val);
+            next.id = key;
+            orderedInstalledSolutions.push(next);
+        });
+        fluid.stableSort(orderedInstalledSolutions, function (a, b) {
+            var idx = gpii.captureTool.appOrdering;
+            function idxOrEnd(i) {
+                return idx.indexOf(i) === -1 ? idx.length + 1 : idx.indexOf(i);
```

Comment:
Why do we send them to the end?
Also, sending them to idx.length would be sufficient.
Also, avoid the double computation

Reply:
If they aren't in the "presentation registry" list of ordered solutions, I put them on the end at the bottom of the page.